File: Pokua/BasicChallenges/Exercise29.java
package BasicChallenges;

import java.util.Scanner;

public class Exercise29{

    public boolean isLeapYear(int year){
        if (year % 4 == 0 && year % 100 != 0 || year % 400 == 0) {
            return true;
        }
        else {
            return false;
        }

    }

    public int daysInMonth(int year, int month) {

        int[] monthDays = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

        if (isLeapYear(year) && month == 2) {
                return 29;
        }

        return monthDays[month - 1];
    }

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Enter a year: ");
        int year = input.nextInt();

        System.out.print("Enter a month: ");
        int month = input.nextInt();

        Exercise29 obj = new Exercise29();
        int days = obj.daysInMonth(year, month);

        System.out.println("Days in the month: " + days);

        input.close();
    }
}

File: KwameAmoah/Exercise29.py
def is_leap(year):
  if year % 4 == 0:
    if year % 100 == 0:
      if year % 400 == 0:
        print("True")
      else:
        print("False")
    else:
      print("True")
  else:
    print("False")

def days_in_month(year, month):
  month_days = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31] 
  a = month_days[1] + 1
  if (year % 4 == 0 or year % 100 == 0 or year % 400 == 0) and (month == 2):
    print(a)
  else:
    print(month_days[month - 1])

  
#🚨 Do NOT change any of the code below 
year = int(input("Enter a year: "))
month = int(input("Enter a month: "))
days = days_in_month(year, month)
print(days)
#Modify the above code to your programming language equivalent but do not change how it works